### PR TITLE
ARTEMIS-2838 - migrate to HawtIO 2

### PR DESCRIPTION
**Pull request by andytaylor**
https://issues.apache.org/jira/browse/ARTEMIS-2838

The main reason of upgrading to HawtIO 2 is that HawtIO 1 is no longer supported for CVE's etc.

Although this is a rewrite all the screens perform the same functionally, although there are improvements and additions.

Refer to the Jira for more info.

Since this is a big change Id be grateful if anyone has the time to review and kick the tyres.

### CODE REVIEW

**Comment by michaelandrepearce** [comment]
Have been able to build and run at first look. Nice. I havent gone through all screens yet to check for issues ill try do that during the week.

File: artemis-hawtio/artemis-plugin/pom.xml [hunk 1]
```diff
@@ -74,6 +78,12 @@
       <artifactId>hawtio-plugin-mbean</artifactId>
     </dependency>
 
+    <!--<dependency>
```
Comment (michaelandrepearce) [inline artemis-hawtio/artemis-plugin/pom.xml:81]: Commented code remove if not needed

**Comment by jbertram** [comment]
I just built the branch and tested it out. When I created my broker is used `--allow-anonymous` and when I "logged in" to the console it just put in a dummy username and password. The console loaded fine, but whenever I clicked on anything in the tree (e.g. addresses, acceptors, etc.) the panel on the right would just flash something quickly and return to the normal status screen with the address size, etc. I figured this was related to security so I stopped the broker and changed my `etc/management.xml` to be:
```xml
<management-context xmlns="http://activemq.org/schema">
</management-context>
```
When I restarted the broker and tried to log back in I got this NPE in the log:
```
2020-09-14 14:11:59,952 ERROR [io.hawt.system.RBACMBeanInvoker] Error while invoking JMXSecurity MBean: javax.management.RuntimeMBeanException: java.lang.NullPointerException: com.google.common.util.concurrent.UncheckedExecutionException: javax.management.RuntimeMBeanException: java.lang.NullPointerException
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2052) [guava-24.1.1-jre.jar:]
	at com.google.common.cache.LocalCache.get(LocalCache.java:3943) [guava-24.1.1-jre.jar:]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3967) [guava-24.1.1-jre.jar:]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4952) [guava-24.1.1-jre.jar:]
	at io.hawt.system.RBACMBeanInvoker.canInvoke(RBACMBeanInvoker.java:215) [hawtio-system-2.10.2.jar:2.10.2]
	at io.hawt.system.RBACRestrictor.isOperationAllowed(RBACRestrictor.java:70) [hawtio-system-2.10.2.jar:2.10.2]
	at org.jolokia.handler.ExecHandler.checkForRestriction(ExecHandler.java:63) [jolokia-core-1.6.2.jar:]
	at org.jolokia.handler.ExecHandler.checkForRestriction(ExecHandler.java:40) [jolokia-core-1.6.2.jar:]
	at org.jolokia.handler.JsonRequestHandler.handleRequest(JsonRequestHandler.java:87) [jolokia-core-1.6.2.jar:]
	at org.jolokia.backend.MBeanServerExecutorLocal.handleRequest(MBeanServerExecutorLocal.java:109) [jolokia-core-1.6.2.jar:]
	at org.jolokia.backend.MBeanServerHandler.dispatchRequest(MBeanServerHandler.java:161) [jolokia-core-1.6.2.jar:]
	at org.jolokia.backend.LocalRequestDispatcher.dispatchRequest(LocalRequestDispatcher.java:99) [jolokia-core-1.6.2.jar:]
	at org.jolokia.backend.BackendManager.callRequestDispatcher(BackendManager.java:429) [jolokia-core-1.6.2.jar:]
	at org.jolokia.backend.BackendManager.handleRequest(BackendManager.java:158) [jolokia-core-1.6.2.jar:]
	at org.jolokia.http.HttpRequestHandler.executeRequest(HttpRequestHandler.java:197) [jolokia-core-1.6.2.jar:]
	at org.jolokia.http.HttpRequestHandler.handlePostRequest(HttpRequestHandler.java:137) [jolokia-core-1.6.2.jar:]
	at org.jolokia.http.AgentServlet$3.handleRequest(AgentServlet.java:460) [jolokia-core-1.6.2.jar:]
	at org.jolokia.http.AgentServlet.handleSecurely(AgentServlet.java:350) [jolokia-core-1.6.2.jar:]
	at org.jolokia.http.AgentServlet.handle(AgentServlet.java:321) [jolokia-core-1.6.2.jar:]
	at org.jolokia.http.AgentServlet.doPost(AgentServlet.java:284) [jolokia-core-1.6.2.jar:]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707) [jetty-all-9.4.27.v20200227-uber.jar:9.4.27.v20200227]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [jetty-all-9.4.27.v20200227-uber.jar:9.4.27.v20200227]
	at org.eclipse.jetty.servlet.ServletHolder$NotAsyncServlet.service(ServletHolder.java:1395)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:755)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1617)
	at io.hawt.web.auth.LoginRedirectFilter.doFilter(LoginRedirectFilter.java:57) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.auth.AuthenticationFilter.lambda$executeAs$1(AuthenticationFilter.java:106) [hawtio-system-2.10.2.jar:2.10.2]
	at java.security.AccessController.doPrivileged(Native Method) [rt.jar:1.8.0_151]
	at javax.security.auth.Subject.doAs(Subject.java:422) [rt.jar:1.8.0_151]
	at io.hawt.web.auth.AuthenticationFilter.executeAs(AuthenticationFilter.java:105) [hawtio-system-2.10.2.jar:2.10.2]
	at io.hawt.web.auth.AuthenticationFilter.doFilter(AuthenticationFilter.java:72) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.filters.HttpHeaderFilter.doFilter(HttpHeaderFilter.java:43) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at io.hawt.web.auth.SessionExpiryFilter.process(SessionExpiryFilter.java:166) [hawtio-system-2.10.2.jar:2.10.2]
	at io.hawt.web.auth.SessionExpiryFilter.doFilter(SessionExpiryFilter.java:60) [hawtio-system-2.10.2.jar:2.10.2]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1596)
	at org.apache.activemq.artemis.component.JolokiaFilter.doFilter(JolokiaFilter.java:50) [artemis-web-2.16.0-SNAPSHOT.jar:2.16.0-SNAPSHOT]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:545)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:590)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1300)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:485)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1215)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:59)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:500)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
	at java.lang.Thread.run(Thread.java:748) [rt.jar:1.8.0_151]
Caused by: javax.management.RuntimeMBeanException: java.lang.NullPointerException
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrow(DefaultMBeanServerInterceptor.java:839) [rt.jar:1.8.0_151]
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrowMaybeMBeanException(DefaultMBeanServerInterceptor.java:852) [rt.jar:1.8.0_151]
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:821) [rt.jar:1.8.0_151]
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801) [rt.jar:1.8.0_151]
	at io.hawt.system.RBACMBeanInvoker.doCanInvoke(RBACMBeanInvoker.java:165) [hawtio-system-2.10.2.jar:2.10.2]
	at io.hawt.system.RBACMBeanInvoker$1.load(RBACMBeanInvoker.java:143) [hawtio-system-2.10.2.jar:2.10.2]
	at io.hawt.system.RBACMBeanInvoker$1.load(RBACMBeanInvoker.java:139) [hawtio-system-2.10.2.jar:2.10.2]
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3524) [guava-24.1.1-jre.jar:]
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2273) [guava-24.1.1-jre.jar:]
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2156) [guava-24.1.1-jre.jar:]
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2046) [guava-24.1.1-jre.jar:]
	... 85 more
Caused by: java.lang.NullPointerException
	at org.apache.activemq.artemis.core.server.management.impl.HawtioSecurityControlImpl.canInvoke(HawtioSecurityControlImpl.java:93) [artemis-server-2.16.0-SNAPSHOT.jar:2.16.0-SNAPSHOT]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_151]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_151]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_151]
	at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71) [rt.jar:1.8.0_151]
	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source) [:1.8.0_151]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_151]
	at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275) [rt.jar:1.8.0_151]
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:112) [rt.jar:1.8.0_151]
	at com.sun.jmx.mbeanserver.StandardMBeanIntrospector.invokeM2(StandardMBeanIntrospector.java:46) [rt.jar:1.8.0_151]
	at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237) [rt.jar:1.8.0_151]
	at com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138) [rt.jar:1.8.0_151]
	at com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252) [rt.jar:1.8.0_151]
	at javax.management.StandardMBean.invoke(StandardMBean.java:405) [rt.jar:1.8.0_151]
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819) [rt.jar:1.8.0_151]
	... 93 more
```

**Comment by jbertram** [comment]
I think I see what the "issue" was when clicking on anything...I was expecting to see the attributes of the object when I clicked on it. However, that information is on the "Attributes" tab. The default tab (i.e. the "Status" tab) continues to display when clicking on anything in the tree. Fair enough.

File: artemis-hawtio/artemis-plugin/pom.xml (hunk 1 above)
Comment (andytaylor) [reply artemis-hawtio/artemis-plugin/pom.xml:81]: fixed

**Comment by andytaylor** [comment]
@jbertram The npe is caused because we dont create a guard if no access list is configured. I'll add a check to only add  the security mbean if this is not null

**Comment by andytaylor** [comment]
> @jbertram The npe is caused because we dont create a guard if no access list is configured. I'll add a check to only add the security mbean if this is not null

This is now fixed

**Comment by michaelpearce-gain** [comment]
A few look and feel bits, on login the brand seems a bit off having some kind of greeny yellow glow, maybe should be the magenta, like wise button be magenta and not hard white background but more transparent

image

compared with

image

**Comment by michaelpearce-gain** [comment]
Next again on brand, something with the fonts seem off, see here it seems to have gone for some times kind of font.

image

where as previous we had a smoother inline with logo - using opensans

image

alot of this was in css here:

https://github.com/apache/activemq-artemis/blob/master/artemis-hawtio/activemq-branding/src/main/webapp/plugin/css/activemq.css

**Comment by michaelpearce-gain** [comment]
functionality wise, everything seems to be there, so its just activemq branding that needs some attention here

**Comment by michaelpearce-gain** [comment]
Lastly, once theming is fixed, i would expect we probably want new screenshots for the docs, and docs upated with new screenshots.,

**Comment by andytaylor** [comment]
I'll take a look at the themes and see if I can cop over and get working all of the old activemq.css. I'll get on this next week.

**Comment by andytaylor** [comment]
Ive gone thru and fixed up the css so magenta is used for hovers and tables etc. Also fixed up the login page and quite a few other areas.

@michaelandrepearce could you give it another look and I can make another pass with any other issues on branding you find.

re the remember me, this isnt possible at the moment but the HawtIO team will fix it and we can update HawtIO once available, its tracked at https://github.com/hawtio/hawtio/issues/2652

**Comment by andytaylor** [comment]
actually re the rememberme, do we really need or want this. HawtIO team think its a bad idea?

**Comment by michaelpearce-gain** [comment]
i cannot speak for all users, but its used where i am working.

**Comment by michaelpearce-gain** [comment]
looks better, i spotted that these new bits, dont seem to be covered and have blue bars.

image

Also some reason the fonts for me still seem odd, and not picking up.

**Comment by michaelpearce-gain** [comment]
as well something is up with the css sizing/spacing with the larger login parts,

notice how there is scroll bars:

=image

and then when i scroll.....
image

i would expect it to fit to screen, i was using my laptop which is a 14inch screened - lenovo T14,

**Comment by andytaylor** [comment]
Ive fixed the blue line on the Broker Info panel.

re the login page, the scroll bar issue was happening because the panel containing the image was larger than the background div. I've removed the 'management console' part of this as this was duplicated underneath to make it smaller. The problem still exists but only for a brief moment before the flow is change to handle the smaller window. I've raised this with the HawtIO team to see if they can fix it, I thin its fine for now tho.

Can you point me to which specific fonts don't look correct, The ones I look at look ok and when i inspect them in the web developer console they show as being OpenSans so Im not sure hwat else I can do with this.

re the remember me checkbox, The hawtIO team don't want to add this, they removed it because it was a security vulnerability and are wondering why the browser's auto fill functionality isn't enough. If this is a blocker I'm not sure how we could proceed, personally I've never used it.

@michaelpearce-gain let me know your thoughts.

**Comment by michaelandrepearce** [comment]
@andytaylor i was looking at keycloak, from the looks of it, theyre using hawt2, may the solution would be to have their layout with the logo above the login?

image

**Comment by andytaylor** [comment]
I've managed to fix the scrollbar issue by resizing the image.

Is there anything else I need to do before this can get merged?

**Comment by michaelpearce-gain** [comment]
@andytaylor been looking at the font thing it really has bugged me, ive found locally if i change a small bit, it seems to look better, i sent you a pr, please do a build yourself and have a look to see if it looks now better?

**Comment by michaelpearce-gain** [comment]
@andytaylor  acutally found a bit more, digging on this, in the activemq.css file, youve added

@font-face {
    font-family: 'PatternFlyIcons-webfont';
    src: url('../../../console/fonts/PatternFlyIcons-webfont.eot');
    url('../../../console/fonts/PatternFlyIcons-webfont.woff') format('woff'),
    url('../../../console/fonts/PatternFlyIcons-webfont.ttf') format('truetype'),
    url('../../../console/fonts/PatternFlyIcons-webfont.svg') format('svg');
    font-weight: normal;
    font-style: normal;
}

@font-face {
    font-family: 'OpenSans';
    src: url('../../../console/fonts/OpenSans-Regular-webfont.eot');
    url('../../../console/fonts/OpenSans-Regular-webfont.woff') format('woff'),
    url('../../../console/fonts/OpenSans-Regular-webfont.ttf') format('truetype'),
    url('../../../console/fonts/OpenSans-Regular-webfont.svg') format('svg');
    font-weight: normal;
    font-style: normal;
}

Yet when build and you run this path doesnt exist these fonts are not present at these paths (looks like not at all) thus why even so CSS is saying thats the font to use, the resource for the font isnt there /hasnt loaded correctly

image

**Comment by andytaylor** [comment]
Ive added your changes @michaelpearce-gain , tested on firefox and chrome and looks fine.

**Comment by michaelpearce-gain** [comment]
i cannot see then any other reason not to merge this.

**Comment by michaelandrepearce** [comment]
will merge it over weekend or monday unless someone beats me to it. If no one else comments

**Comment by andytaylor** [comment]
@michaelpearce-gain thanks for your help with reviewing this, much appreciated

**Comment by michaelandrepearce** [comment]
@andytaylor no worries. One thing i just realised i havent checked was docs, did the docs get updated with new screenshots?

**Pull request closed without merging**